### PR TITLE
Enable adress translator

### PR DESCRIPTION
**Pull request by MALPI**
I added a factory to enable configuration of [AddressTranslators](https://docs.datastax.com/en/drivers/java/2.0/com/datastax/driver/core/policies/EC2MultiRegionAddressTranslater.html).

I therefore created a new `connect` package since it didn't seem to fit anywhere else.

See Issue #25

### CODE REVIEW

File: src/main/java/systems/composable/dropwizard/cassandra/CassandraFactory.java [hunk 1]
```diff
@@ -201,6 +207,9 @@
     @Valid
     private Optional<PoolingOptionsFactory> poolingOptions = Optional.empty();
 
+    @Valid
+    private Optional<AddressTranslatorFactory> translatorOptions = Optional.empty();
```
Comment (stuartgunter) [inline src/main/java/systems/composable/dropwizard/cassandra/CassandraFactory.java]: Please can we call this `addressTranslator` instead.

File: src/main/java/systems/composable/dropwizard/cassandra/CassandraFactory.java [hunk 2]
```diff
@@ -451,6 +471,7 @@ public Cluster build(MetricRegistry metrics, HealthCheckRegistry healthChecks) {
             builder.addContactPoints(contactPoint);
         }
 
+
```
Comment (stuartgunter) [inline src/main/java/systems/composable/dropwizard/cassandra/CassandraFactory.java]: Please don't add extra whitespace. The PR should conform to the existing code styles used.

File: src/main/java/systems/composable/dropwizard/cassandra/CassandraFactory.java [hunk 3]
```diff
@@ -465,6 +486,8 @@ public Cluster build(MetricRegistry metrics, HealthCheckRegistry healthChecks) {
         queryOptions.ifPresent(builder::withQueryOptions);
         socketOptions.ifPresent(builder::withSocketOptions);
         poolingOptions.map(PoolingOptionsFactory::build).ifPresent(builder::withPoolingOptions);
+        translatorOptions.map(AddressTranslatorFactory::build).ifPresent(builder::withAddressTranslator);
+
```
Comment (stuartgunter) [inline src/main/java/systems/composable/dropwizard/cassandra/CassandraFactory.java]: Please don't add extra whitespace. The PR should conform to the existing code styles used.

File: src/main/java/systems/composable/dropwizard/cassandra/connect/AddressTranslatorFactory.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ *   <li>Add a {@code META-INF/services/systems.composable.dropwizard.cassandra.connect.AddressTranslatorFactory} file with your implementation's full class
+ *       name to the class path.
+ * </ol>
+ *
+ * @see EC2MultiRegionAddressTranslatorFactory
+ * @see IdentityTranslatorFactory
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+public interface AddressTranslatorFactory extends Discoverable {
+  AddressTranslator build();
```
Comment (stuartgunter) [inline src/main/java/systems/composable/dropwizard/cassandra/connect/AddressTranslatorFactory.java]: Please use the same indentation (4 spaces, not 2) as existing code to maintain uniformity. The same applies to other code in this PR.

File: src/main/java/systems/composable/dropwizard/cassandra/connect/EC2MultiRegionAddressTranslatorFactory.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package systems.composable.dropwizard.cassandra.connect;
+
+import com.datastax.driver.core.policies.AddressTranslator;
+import com.datastax.driver.core.policies.EC2MultiRegionAddressTranslator;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+
+/** A factory for configuring and building {@link com.datastax.driver.core.policies.EC2MultiRegionAddressTranslator} instances. */
+@JsonTypeName("ec2MultiRegionAddressTranslator")
```
Comment (stuartgunter) [inline src/main/java/systems/composable/dropwizard/cassandra/connect/EC2MultiRegionAddressTranslatorFactory.java]: Please rename to `ec2MultiRegion` (to match naming formats in use for other factories)... the `AddressTranslator` part is redundant.

File: src/main/java/systems/composable/dropwizard/cassandra/connect/IdentityTranslatorFactory.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package systems.composable.dropwizard.cassandra.connect;
+
+import com.datastax.driver.core.policies.AddressTranslator;
+import com.datastax.driver.core.policies.IdentityTranslator;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+
+/** A factory for configuring and building {@link com.datastax.driver.core.policies.IdentityTranslator} instances. */
+@JsonTypeName("identityTranslator")
```
Comment (stuartgunter) [inline src/main/java/systems/composable/dropwizard/cassandra/connect/IdentityTranslatorFactory.java]: What do we gain by defining this type? It looks like its equivalent to not specifying an `addressTranslator`.

File: src/test/java/systems/composable/dropwizard/cassandra/connect/Ec2MultiRegionAddressTranslatorFactoryTest.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
+package systems.composable.dropwizard.cassandra.connect;
+
+import org.junit.Test;
+
+import static org.hibernate.validator.internal.util.Contracts.assertNotNull;
+
+public class Ec2MultiRegionAddressTranslatorFactoryTest {
```
Comment (stuartgunter) [inline src/test/java/systems/composable/dropwizard/cassandra/connect/Ec2MultiRegionAddressTranslatorFactoryTest.java]: Please capitalise `EC2` (i.e. no lowercase `c`) to conform to existing naming conventions.

File: src/test/java/systems/composable/dropwizard/cassandra/connect/IdentityTranslatorFactoryTest.java [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package systems.composable.dropwizard.cassandra.connect;
+
+import org.junit.Test;
+
+import static org.hibernate.validator.internal.util.Contracts.assertNotNull;
+
+public class IdentityTranslatorFactoryTest {
+  @Test
+  public void buildsPolicyWithChildPolicy() throws Exception {
```
Comment (stuartgunter) [inline src/test/java/systems/composable/dropwizard/cassandra/connect/IdentityTranslatorFactoryTest.java]: Please define a more suitable name for this test.

File: src/test/java/systems/composable/dropwizard/cassandra/connect/Ec2MultiRegionAddressTranslatorFactoryTest.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+  @Test
+  public void buildsPolicyWithChildPolicy() throws Exception {
```
Comment (stuartgunter) [inline src/test/java/systems/composable/dropwizard/cassandra/connect/Ec2MultiRegionAddressTranslatorFactoryTest.java]: Please define a more suitable name for this test.

File: src/main/java/systems/composable/dropwizard/cassandra/connect/AddressTranslatorFactory.java [hunk 10]
```diff
@@ -0,0 +1,26 @@
+package systems.composable.dropwizard.cassandra.connect;
```
Comment (stuartgunter) [inline src/main/java/systems/composable/dropwizard/cassandra/connect/AddressTranslatorFactory.java]: Please could you rename this package... `connect` is a verb, which doesn't fit nicely with package names. Perhaps something like `connection` or `network`?

**Review by stuartgunter — CHANGES_REQUESTED** [review]
Thanks for the PR - it's a great help!

I've added some comments - primarily to keep the coding style consistent. Please could you take a look and either fix / add your own comments?

Much appreciated.

**Comment by MALPI** [comment]
@stuartgunter I applied all changes you mentioned. Thanks for your quick response.

I am really happy contribute!

**Review by stuartgunter — APPROVED** [review]

**Comment by MALPI** [comment]
@stuartgunter when will a new release be triggered?

**Comment by stuartgunter** [comment]
@MALPI I have one other PR that I'd like to merge before doing the next release, but it should go out in the next few days.

**Comment by MALPI** [comment]
@stuartgunter any update?

**Comment by stuartgunter** [comment]
I released a new version yesterday, so you should see it in Maven Central now. Please let me know if you have any problems with the latest version.

**Pull request merged**